### DISCUSSION TITLE
New users issues not answered in FAQ

### DISCUSSION THREAD

**Discussion by demianph** [automatic-ripping-machine/automatic-ripping-machine#544] [General]
I am coming from doing windows encoding by hand to ARM. My day job is Linux/AIX/*Unix admin work so moving to /setting up ARM is more of a shift in workflow than concept.

After a LOT of trial and error (different distros, different releases of ubuntu, hardware failures) I finally got ARM installed in Ubuntu 20.04.4 LTS.

I have a i3-7100 CPU with no internal GPU so no quick sync and an old nvidia card (9 series) so not using nvenc. CPU encoding only, which I do expect to be slower.
HDD is a 1TB WD black and all devices are on SATA connections (6Gb max). Finished media is offloaded to main storage server for jellyfin and navidrome to consume.
Have a LG DVD drive for CD/DVD media
Picked up an ASUS U4K DB drive which I lucked out on as it came with unencrypted firmware doe Blu Ray.

ARM works. 

I found issues with setup in a few areas. More than happy to re-do the process and document any differences for updating install instructions.

One area I really got into the weeds on was abcde configuration.
It kept erroring in way that were unintuitive, turns out ubuntu depreciated the app that abcde wants to use to normalize mp3 (but not flac) and replaced it with a different binary.

I started testing once I disabled mp3 normalization option in the config.

CD ripping via cd/dvd drive checks out ok, data seems to be as expected at the end.
DVD ripping via same drive and encoding speed checks out ok, I tried both tv series and movies. Movies seems to be getting what looks like a duplicate of the main video file in extras (commentary version) going by size?
Blu Ray, I have a dedicated drive for this. First disc I put in rips quite fast but something did not go as expected in the transcode.
It took 16 hours and 8 minutes and the directory size in the completed folder is 23 GB. 6GB for main video and 17GB in the extras folder. Like with DVD movies I see what appears to be a duplicate of the main video in extras.

Have I failed to configure something correctly?
I wanted to ask about basic settings I may have overlooked first before doing this again in debug mode and posting logs.

**Comment by 1337-server** [comment]
> I have a i3-7100 CPU with no internal GPU so no quick sync and an old nvidia card (9 series) so not using nvenc. CPU encoding only, which I do expect to be slower.

Depending on your desired output, it's physically possible to use hardware transcoding on the CPU you have. This would speed things up massively. But is a massive quality hit for Blu-ray. It's barely noticeable IMO for DVD.
I tested this on an i3-6100 with h.264 Intel QuickSync (Docker image isn't ready for HW transcoding yet)

> DVD ripping via same drive and encoding speed checks out ok, I tried both tv series and movies. Movies seems to be getting what looks like a duplicate of the main video file in extras (commentary version) going by size?

If you are using MakeMKV with Series, MakeMKV will dump a full combined version. I haven't found a way to disable this behavior. For movies the only way to remove this is to enable to mainfeature, but this also remove all extras.
 
> Blu Ray, I have a dedicated drive for this. First disc I put in rips quite fast but something did not go as expected in the transcode.
It took 16 hours and 8 minutes and the directory size in the completed folder is 23 GB. 6GB for main video and 17GB in the extras folder. Like with DVD movies I see what appears to be a duplicate of the main video in extras.

Sadly, this sounds roughly about correct, with some better profiles (4K, the best quality, etc.) for Blu-ray for your hardware. But this heavily depends on the profile you have selected.


> One area I really got into the weeds on was abcde configuration.
It kept erroring in way that were unintuitive, turns out ubuntu depreciated the app that abcde wants to use to normalize mp3 (but not flac) and replaced it with a different binary.

> I started testing once I disabled mp3 normalization option in the config.

Could you go into more detail on this ?

**Reply by demianph** [reply]
>> One area I really got into the weeds on was abcde configuration.
It kept erroring in way that were unintuitive, turns out ubuntu depreciated the app that abcde wants to use to normalize mp3 (but not flac) and replaced it with a different binary.

>> I started testing once I disabled mp3 normalization option in the config.

> Could you go into more detail on this ?

It appears that as of ubuntu 15.04. mp3gain was removed, so if you place replaygain in your abcde config ACTIONS= line it says it can not find the app mp3gain. 

Looks like since 18.04 LTS the application to handle replaygain for mp3 is called replaygain and seems to be python related as you install it with apt-get install python-rgain

Gained my info from these pages.
[https://askubuntu.com/questions/1189418/automated-ripping-machine-arm-ripping-to-flac-not-ripping-to-mp3](https://askubuntu.com/questions/1189418/automated-ripping-machine-arm-ripping-to-flac-not-ripping-to-mp3)
[https://askubuntu.com/questions/246242/how-to-normalize-sound-in-mp3-files](https://askubuntu.com/questions/246242/how-to-normalize-sound-in-mp3-files)
[https://manpages.ubuntu.com/manpages/focal/man1/replaygain.1.html](https://manpages.ubuntu.com/manpages/focal/man1/replaygain.1.html)

**Reply by demianph** [reply]
I also somehow was missing libbluray and had to add that once I saw the messages in my test output.

 _BD-J check: Failed to load libbluray.jar_

**Reply by demianph** [reply]
Correction, I still get that error.

I see 
libbluray-j2se-1.2.0.jar
in /usr/share/java

```
arm@arm:~/logs$ sudo apt search libbluray
Sorting... Done
Full Text Search... Done
libbluray-bdj/focal,now 1:1.2.0-1 all [installed]
  Blu-ray Disc Java support library (BD-J library)
```

**Reply by 1337-server** [reply]
> I also somehow was missing libbluray and had to add that once I saw the messages in my test output.
 BD-J check: Failed to load libbluray.jar

https://github.com/HandBrake/HandBrake/issues/874

This should give more info


>Gained my info from these pages.
https://askubuntu.com/questions/1189418/automated-ripping-machine-arm-ripping-to-flac-not-ripping-to-mp3

This is great, thanks. So it's as simple as removing `replaygain` from the abcde.conf ?
I'm not sure of the best solution as I don't use mp3's

**Reply by demianph** [reply]
Removing that means you have to do your mp3 replaygain work outside of ARM. FLAC replaygain still works, you specify it in the flac options rather than the ACTIONS= setting (and handles by different software).
I'm not sure how good I could be but maybe I can see how we can get it to detect ubuntu versions and have it choose the available utility for replaygain based on that.

**Comment by demianph** [comment]
Thanks for the info, my expectations are set for what I am telling this to do.
I did hit a couple points where additional packages are needed but I was in GSD mode so didn't take notes.
Some scripts and linked items in various instructions on the wiki to github ARM resources return 404 when clicked.

I am willing to run through the setup process and documentation again if that would help.

**Reply by 1337-server** [reply]
>  Some scripts and linked items in various instructions on the wiki to github ARM resources return 404 when clicked.

Do you know which links are broken ?

> I am willing to run through the setup process and documentation again if that would help.

That would be a great help, thanks.

**Reply by demianph** [reply]
Will do. You might not hear back from me until mid August, due to life, but I will give it a thorough run through.
